### DISCUSSION TITLE
"Drafts" or hidden documents

### DISCUSSION THREAD

**Discussion by vcavallo** [srid/emanote#565] [Ideas]
Preamble: I recognize that emanote is not meant to be a publishing platform and thus this may be scope creep.

When collecting thoughts and writing stub pages, it would be helpful to be able to mark these documents as "drafts" which would be visible in the live dev server site but skipped upon building (similarly to how broken links and the "little dev server status dot" are rendered on the dev site but not in the built site).

This would allow me to keep all my writing _in_ my emanote project without having to move files in and out as they graduate to full notes.  
Perhaps there is a way to do this with the .yaml files that i haven't yet come across? Like having a directory for "drafts" that is somehow not included in the build - maybe using `layers` (which I've never looked into).

I'm sure there are complications that would arise from this; the ones that come to mind right away:

- If a `draft` page is linked to from a `live` page, is this a broken link? does it get displayed but somehow "disabled" in the built UI?
- Does this become a top-level feature of emanote, and if so, how to make it backwards compatible for those who don't plan to use it?

**Comment by srid** [comment]
You could add the drafts folder as a separate layer only in dev server runs, and skip that when doing static site build. Also, this layer can be [mounted](https://emanote.srid.ca/guide/layer#mount-point) as a sub-path of URL (as of #523).

```
# live server
emanote -L ./notes;./drafts run

# static site gen
emanote -L ./notes gen
```

Since the more general "layers" feature can solve this, I see little need to add the specific feature. 

Yes, linking would be shown as broken link during static site gen. That's a feature, right? It would force you to address those broken links before publishing.

**Comment by vcavallo** [comment]
good points. I'll have to look into the layers feature more to accomplish this, thanks.

[Update: he checked, it worked, thanks!]

One minor detail: when using the emanote gh template, one cannot pass args to the `nix run` command, right?
